### PR TITLE
add completion() for ollama

### PR DESCRIPTION
**Pull request by dineshyv**
This PR implements the completion API for ollama adapter.

Test plan:

PROVIDER_ID=test-ollama PROVIDER_CONFIG=tests/inference/provider_config_example.yaml MODEL_IDS="Llama3.2-1B-Instruct" pytest -s tests/inference/test_inference.py --tb=short --disable-warnings -rs

PROVIDER_ID=meta-reference PROVIDER_CONFIG=tests/inference/provider_config_example.yaml MODEL_IDS="Llama3.2-1B-Instruct" pytest -s tests/inference/test_inference.py --tb=short --disable-warnings -rs

Screenshot 2024-10-21 at 10 07 27 PM

### CODE REVIEW

File: llama_stack/providers/tests/inference/test_inference.py [hunk 1]
```diff
@@ -132,7 +132,7 @@ async def test_completion(inference_settings):
     params = inference_settings["common_params"]
 
     provider = inference_impl.routing_table.get_provider_impl(params["model"])
-    if provider.__provider_id__ != "meta-reference":
+    if provider.__provider_id__ not in ("meta-reference", "test-ollama"):
```
Comment (ashwinb) [inline llama_stack/providers/tests/inference/test_inference.py:135]: ah my bad. this check should be based on `__provider_type__` and not `__provider_id__` 

can you change it to:

```
if provider.__provider_type__ not in ("meta-reference", "remote::ollama"):
```

**Review by ashwinb — APPROVED** [review]
Suhweeet! One inline comment.

File: llama_stack/providers/tests/inference/test_inference.py (hunk 1 above)
Comment (ashwinb) [reply llama_stack/providers/tests/inference/test_inference.py:135]: actually `provider.__provider_type__` is not quite right. hold on.

Comment (ashwinb) [reply llama_stack/providers/tests/inference/test_inference.py:135]: Should be `provider.__provider_spec__.provider_type`

Comment (dineshyv) [reply llama_stack/providers/tests/inference/test_inference.py]: done!

**Pull request merged**